pkg/gofr/container/container.go

```diff
@@ -1,3 +1,6 @@
+// The container package provides a centralized structure to manage common application-level concerns such as
+// logging, connection pools, and service management. This package is designed to facilitate the sharing and
```

Comment:
can you also include data sources which comprises of different databases, pubsub, filesystem etc.

Reply:
Sure. Also, I have a question about documentation in go. I saw that `package container` is used in many files which made me wonder what is the best way to document such package? How do I choose which file to add documentation?

Reply:
`container.go` is the central package which also has the container struct. So it is the best file.

Reply:
I wanted to add some references but couldn't figure out how exactly it works in godoc 😄 
```
Supported data sources:
  - Databases ([cassandra], [clickhouse], [mongo], [solr], [dgraph], [sql])
  - Key value storages [redis] [kafka]
  - Pub/Sub systems ([google], [kafka], [mqtt])
  - File system [file]
```
Results:
<img width="569" alt="image" src="https://github.com/user-attachments/assets/239447ac-9978-4904-a06e-b4290965108b">

Even more links to other packages generate false(not found) URL at least locally
Can you help me with this?

Reply:
You may refer to the following documentation 
https://tip.golang.org/doc/comment#links